sound/soc/sof/intel/pci-apl.c

```diff
@@ -29,6 +29,7 @@ static const struct sof_dev_desc bxt_desc = {
 	.chip_info = &apl_chip_info,
 	.ipc_supported_mask	= BIT(SOF_IPC) | BIT(SOF_INTEL_IPC4),
 	.ipc_default		= SOF_IPC,
+	.dspless_mode_supported	= true,		/* Only supported for HDaudio */
```

Comment:
this could also be set with the test in the interface version. There's no good reason why we set the interfaces with a function and this flag as a constant

Reply:
Yeah, but I have not been able to get my hands on on other than TGL and APL class device. It should work on others, but I can not say for sure...
But then we will have issue in core to block the dspless mode on platforms that it is _known_ to not work (AMD, iMX, MTK, Atom).
We should have this flag.

Reply:
ok but then let's add it everywhere, not just APL and TGL

Reply:
Hrm, HDA devices should work just fine in DSPless mode. I guess I can flag them even if I was not able to run the DSPless mode on them?